code/__HELPERS/unsorted.dm

```diff
@@ -1214,3 +1214,24 @@ proc/shift_proximity(atom/checker, atom/A, range, atom/B, newrange)
 		var/turf/F = E
 		F.proximity_checkers |= checker
 	return 1
+
+/proc/flick_overlay_better(image/I, atom/A, duration) //todo :^) change flick_overlay() to this
+	set waitfor = 0
+	if(!A || !I)
+		return
+	A.overlays |= I
+	sleep(duration)
```

Comment:
It can remain as an alternate proc that only acts on 1 atom, but it is not becoming the default.
My previous point about sleep --> spawn was erroneous, as you're working with overlays, which can't be animated. which is something this proc can't do that the current implementation can.
so you're not replacing the current one, but yours can stay (if renamed of course, it's not "better")